### PR TITLE
Bug 2063570: ipv6 spoke cluster installation would fail when unchecking 'use advanced network', due to ipv4 defaults cidrs

### PR DESCRIPTION
**Pull request by ori-amizur**
To avoid cases that cluster installation fails due to
network address families of (MachineNetworks, ServiceNetworks, ClusterNetworks) not the same,
a new cluster validation is added.
The validation compares the address families and return failure if not
the same.
In case machine networks are not required (user-managed-networking),
only service and cluster networks are compared.

## List all the issues related to this PR

- [ ] New Feature 
- [x] Enhancement 
- [x] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] Unit tests

## Assignees

/cc @filanov 
/cc @avishayt 

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/cluster/validator.go [hunk 1]
```diff
@@ -569,6 +570,65 @@ func (v *clusterValidator) printNoCidrsOverlapping(c *clusterPreprocessContext,
...
+func (v *clusterValidator) printIsNetworksSameAddressFamilies(context *clusterPreprocessContext, status ValidationStatus) string {
+	switch status {
+	case ValidationFailure:
+		var networks string
+		if network.IsMachineNetworkRequired(context.cluster) {
+			networks = "(MachineNetworks, ServiceNetworks, ClusterNetworks)"
+		} else {
+			networks = "(ServiceNetworks, ClusterNetworks)"
+		}
+		return fmt.Sprintf("Address families of networks %s is not the same.", networks)
```
Comment (avishayt) [inline internal/cluster/validator.go:617]:
```suggestion
		return fmt.Sprintf("Address families for networks %s are not the same.", networks)
```

**Review by avishayt — APPROVED** [review]

**Comment by filanov** [comment]
will need to backport please retitle to the BZ

**Comment by ori-amizur** [comment]
/retitle  Bug 2063570: ipv6 spoke cluster installation would fail when unchecking 'use advanced network', due to ipv4 defaults cidrs

**Review by filanov — APPROVED** [review]

**Pull request merged**